pkg/cluster/condition.go

```diff
@@ -113,6 +116,11 @@ func (m *manager) aroCredentialsRequestReconciled(ctx context.Context) (bool, er
 
 	u, err := m.dynamiccli.Resource(CredentialsRequestGroupVersionResource).Namespace("openshift-cloud-credential-operator").Get(ctx, "openshift-azure-operator", metav1.GetOptions{})
 	if err != nil {
+		// If the CredentialsRequest is not found, it may have just recently been reconciled.
+		// Return nil to retry until we hit the condition timeout.
+		if credentialsRequestNotFoundErrorRegex.MatchString(err.Error()) {
```

Comment:
Use [`errors.IsNotFound()`](https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors#IsNotFound) here (we import this as `kerrors` elsewhere).

```suggestion
		if kerrors.IsNotFound(err) {
```